webui/react/src/components/NotebookModal.tsx

```diff
@@ -0,0 +1,235 @@
+      } else {
+        launchNotebook(
+          undefined,
+          resourceType === ResourceType.CPU ? 0 : values.slots,
+          values.template === 'default' ? undefined : values.template,
+          values.name,
+          values.pool,
+        );
+      }
+      if (onLaunch) onLaunch();
+    },
+    [ showFullConfig, form, onLaunch, resourceType ],
+  );
+
+  const handleResourcePoolUpdate = useCallback((selectedPoolName: string) => {
+    const selectedPool = resourcePools.find(pool => pool.name === selectedPoolName);
+    if (!selectedPool) return;
+    const hasCPUCapacity = selectedPool.cpuContainerCapacityPerAgent > 0;
+    const hasGPUCapacity = selectedPool.slotsPerAgent && selectedPool.slotsPerAgent > 0;
+    if (hasCPUCapacity && hasGPUCapacity) {
+      setShowResourceType(true);
+    } else if (hasCPUCapacity) {
+      setResourceType(ResourceType.CPU);
+      setShowResourceType(false);
+    } else if (hasGPUCapacity) {
+      setResourceType(ResourceType.GPU);
+      setShowResourceType(false);
+    }
+  },[ resourcePools ]);
+
+  const handleTypeUpdate = useCallback((selectedResourceType) => {
+    setResourceType(selectedResourceType as ResourceType);
+  },[]);
+
+  return <Modal
+    footer={<>
+      <Button onClick={handleSecondary}>{showFullConfig ? 'Edit Form' : 'Edit Full Config'}</Button>
+      <Button
+        type="primary"
+        onClick={async () => {
+          try {
+            const values = await form.validateFields();
+            handleCreateEnvironment(values);
+          } catch (e) {}
+        }
+        }>Launch</Button>
+    </>}
+    title="Launch JupyterLab"
+    visible={visible}
+    width={540}
+    {...props}>
+    {showFullConfig ?
+      <Form form={form}>
+        <div className={css.note}>
+          <Link external path="/docs/reference/command-notebook-config.html">
+          Read about notebook settings
+          </Link>
+        </div>
+        <React.Suspense fallback={<div className={css.loading}><Spinner /></div>}>
+          <Item
+            name="config"
+            noStyle
+            rules={[ { message: 'Invalid YAML', required: true }, () => ({
+              validator(_, value) {
+                try {
+                  yaml.load(value);
+                  return Promise.resolve();
+                } catch(err) {
+                  return Promise.reject(new Error('Invalid YAML'));
+                }
+              },
+            }) ]}>
+            <MonacoEditor
+              height={430}
+              language="yaml"
+              options={{
+                minimap: { enabled: false },
+                scrollBeyondLastLine: false,
+                selectOnLineNumbers: true,
+              }} />
+          </Item>
+        </React.Suspense>
+      </Form> :
+      <Form
+        form={form}
+        initialValues={storage.getWithDefault(STORAGE_KEY, {
+          slots:1,
+          template: 'default',
```

Comment:
**chore:** this be `undefined` for the initial value. `default` may not actually map to a resource pool?

![2021-05-25 at 10 39 AM](https://user-images.githubusercontent.com/220971/119535748-967b7b00-bd45-11eb-833d-f6b8bf29b360.png)